core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -37,18 +37,18 @@ import scala.concurrent.duration.Duration
 import scala.concurrent.{Await, Future}
 
 //TODO: add interface to query master here
-class ClientContext(config: Config) {
+class ClientContext(config: Config, sys:Option[ActorSystem], mster: Option[ActorRef]) {
   private val LOG: Logger = LogUtil.getLogger(getClass)
   private implicit val timeout = Timeout(5, TimeUnit.SECONDS)
 
   private val masters = config.getStringList(Constants.GEARPUMP_CLUSTER_MASTERS).toList.flatMap(Util.parseHostList)
 
-  implicit val system = ActorSystem(s"client${Util.randInt}" , config)
+  implicit val system = sys.getOrElse(ActorSystem(s"client${Util.randInt}" , config))
```

Comment:
Instead of doing tricky Option here, how about adding a explicit parameter system: ActorSystem in constructor. 

We can create a default Actorsystem in companion apply if not provided.